### PR TITLE
fix: batch status responsiveness

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
fix: batch status responsiveness

## Motivation and Context
'batch status' was not working correctly
for multiple situations, this PR fixes
that.

## How to test this PR?
`mc batch status` with your favorite job and let
that job fails.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fmc%2fpull%2fNNNNN)

### CODE REVIEW

**Comment by harshavardhana** [comment]
this PR also fixes non-working `--json` mode.

**Comment by harshavardhana** [comment]
PTAL @vadmeste @poornas @krishnasrinivas

**Comment by harshavardhana** [comment]
Without this PR batch status doesn't work with batch key rotation.

File: cmd/batch-status.go [hunk 1]
```diff
@@ -61,25 +62,41 @@ func mainBatchStatus(ctx *cli.Context) error {
 	defer cancel()
 
 	_, e := client.DescribeBatchJob(ctxt, jobID)
-	fatalIf(probe.NewError(e), "Unable to lookup job status")
+	if madmin.ToErrorResponse(e).Code != "XMinioAdminNoSuchJob" {
```
Comment (krisis) [inline cmd/batch-status.go:65]:
```suggestion
	if e != nil && madmin.ToErrorResponse(e).Code != "XMinioAdminNoSuchJob" {
```
The else block too may require similar changes.

**Review by poornas — APPROVED** [review]

**Review by krisis — APPROVED** [review]

**Pull request merged**